src/async_nbgrader/async_nbgrader/apps/processmessageapp.py

```diff
@@ -0,0 +1,86 @@
+# coding: utf-8
+
+import base64
+import json
+import os
+import traceback
+
+from traitlets import default
+from nbgrader.apps.baseapp import NbGrader
+from nbgrader.coursedir import CourseDirectory
+from ..helpers import chdir
+from ..helpers import get_nbgrader_api
+
+
+aliases = {
+    "log-level": "Application.log_level",
+    "db": "CourseDirectory.db_url",
+}
+flags = {}
+
+
+# JupyterHub settings
+JUPYTERHUB_API_URL = os.environ.get("JUPYTERHUB_API_URL") or "http://hub:8081/hub/api"
+JUPYTERHUB_BASE_URL = os.environ.get("JUPYTERHUB_BASE_URL") or "/"
+
+
+class ProcessMessageApp(NbGrader):
+    """App to handle amqp messages from Argo"""
+
+    name = "async_nbgrader-process-message"
+
+    aliases = aliases
+
+    @default("classes")
+    def _classes_default(self):
+        classes = super(ProcessMessageApp, self)._classes_default()
+        classes.append(ProcessMessageApp)
+        return classes
+
+    def start(self):
+        """Handler for processing mesage, message is passed as first argument, it is received in self.extra_args[0]"""
+        super(ProcessMessageApp, self).start()
+        if len(self.extra_args) == 0 or self.extra_args[0] == "":
+            self.fail("message is missing")
+        message = json.loads(self.extra_args[0])
+        encoded_message = message["data"]
+        self.log.info("message " + encoded_message)
+        missing_padding = len(encoded_message) % 4
+        if missing_padding > 0:
+            encoded_message += b"=" * (4 - missing_padding)
+        body = json.loads(base64.b64decode(encoded_message).decode("utf-8")).get("body")
+        if body == None:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The `==` and `!=` operators use the compared objects&#39; ` __eq__` method to test if they are equal. To check if an object is a singleton, such as `None`, we recommend that you use the `is` identity comparison operator.

[Learn more](https://docs.python.org/3/reference/expressions.html#is)